### ISSUE TITLE
Revamp comment.sh shell script

### ISSUE THREAD

**Issue by shwsun** [haoel/leetcode#138]
Ever since leetcode org changed their website, the regx in the `comment.sh` is kinda broken. Any idea to fix it?

**Comment by haoel** [comment]
They use the GraphQL API, I will update my script soon.  

very soon......

**Comment by shwsun** [comment]
Hi, I think there are only two slight problems:     
1. in the `query_problem.sh` we should use `-s` instead of `-q` since that is already deprecated.    
2. the question description is not really inserted into the `source_file`, I think the ` ${TMP_FILE}` should be `${source_file}`.

**Comment by haoel** [comment]
for 1), you mean the option for `xidel` command should use `-s` rather than `-q` ? (the `xidel` acutally hasn't `-s` option)

for 2), the `${TMP_FILE}` is a temporary file which will be insert into `$source_file` late.  

```
#insert the problem description into the source file, and remove it
sed -i.bak '4 r '${TMP_FILE}'' ${source_file}
```

may I know more details?

**Comment by haoel** [comment]
@shwsun  any feedback?

**Comment by shwsun** [comment]
@haoel I am pretty sure about the param problem. Is it because of the newer version of `xidel`?
```
[~/repos/leetcode/scripts]-[master↓]── - xidel -q
-quiet,-q is outdated. Use --silent,-s
```
Re the other issue, the `comment.sh` still can't get the problem description into the file from my side. 
```
[~/repos/leetcode/scripts]-[master↓]── - ./comments.sh https://leetcode.com/problems/largest-number/

-quiet,-q is outdated. Use --silent,-s
-quiet,-q is outdated. Use --silent,-s
-quiet,-q is outdated. Use --silent,-s
-quiet,-q is outdated. Use --silent,-s
-quiet,-q is outdated. Use --silent,-s
./comments.sh: line 85: [: ==: unary operator expected
Create a new file - LargestNumber.cpp.
LargestNumber.cpp updated !
[~/repos/leetcode/scripts]-[master↓]── - cat LargestNumber.cpp 
// Source : https://leetcode.com/problems/largest-number/
// Author : shwsun
// Date   : 2018-02-05

/*************************************************************************************** 
 *
 * 
 ***************************************************************************************/
```
I am running Ubuntu, could you get a Linux and test it a bit?

**Comment by haoel** [comment]
Thanks for more information. 

Yes, the script I only tested on MacOS. And on linux platform, `xidel` rename the `-q` option to `-s` after version 0.9.4, so I need to check the options.

Now, I fix it in commit [0fdf100](https://github.com/haoel/leetcode/commit/0fdf1005493af2397d8ad381996590c6d6054d9e)

**Comment by haoel** [comment]
I am going to close this issue, @shwsun you can reopen it if anything's still problem
